pkg/apis/bindings/v1alpha1/kafka_types.go

```diff
@@ -56,6 +56,10 @@ type KafkaSASLSpec struct {
 	// Password is the Kubernetes secret containing the SASL password.
 	// +optional
 	Password SecretValueFromSource `json:"password,omitempty"`
+
+	// Type of saslType, defaults to plain (vs SCRAM-SHA-512 or SCRAM-SHA-256)
+	// +optional
+	Type SecretValueFromSource `json:"saslType,omitempty"`
```

Comment:
should we name it `type` instead of `saslType` ?  On the source it's more strongly typed / structured - since you are on the `.SASL` path here already